### PR TITLE
Add entry SvidStore to SPIRE Server

### LINKED ISSUE #2069: Serverless support: validation of entries
Research the need to implement some sort of validation of the entries that indicate that the issued identity must be stored externally.

### PR DESCRIPTION
**Pull request by MarcosDY**
Add support to SPIRE Server to storable entries, that has a new boolean "store_svid" (as default all entries are not storable)
Storable entries must have at least 1 selector, and all selector types of that entry must be the same (the plugin that is used to store generated SVIDs on agent side)

**Which issue this PR fixes**
fixes #2069

### CODE REVIEW

File: pkg/server/datastore/sqlstore/sqlstore.go [hunk 1]
```diff
@@ -3194,6 +3216,19 @@ func validateRegistrationEntry(entry *common.RegistrationEntry) error {
 		return sqlError.New("invalid registration entry: missing selector list")
 	}
 
+	// In case of StoreSvid is set, all entries 'must' be the same type,
+	// it is done to avoid users to mix selectors from different platforms in
+	// entries with storable SVIDs
+	if entry.StoreSvid {
+		// Selectors must never be empty
+		tpe := entry.Selectors[0].Type
```
Comment (azdagron) [inline pkg/server/datastore/sqlstore/sqlstore.go:3224]: Can we use the equalSelectorTypes helper here?

**Review by azdagron — COMMENTED** [review]
Looks good! One suggestion...

File: pkg/server/datastore/sqlstore/sqlstore.go (hunk 1 above)
Comment (MarcosDY) [reply pkg/server/datastore/sqlstore/sqlstore.go:3224]: It is not possible, the selector we use here is from `proto common` (https://github.com/spiffe/spire/blob/main/proto/spire/common/common.pb.go#L340)
But equalSelectorTypes is done for `sqlstore/models` (https://github.com/spiffe/spire/blob/main/pkg/server/datastore/sqlstore/models.go#L80)

Comment (azdagron) [reply pkg/server/datastore/sqlstore/sqlstore.go:3224]: Ah ok.

**Review by azdagron — APPROVED** [review]

**Review by amartinezfayo — APPROVED** [review]
LGTM! Thanks @MarcosDY!

**Pull request merged**